### PR TITLE
2887 - 18A REC_OASDI_INSURANCE not mandatory

### LINKED ISSUE #2887: as tech lead, I need SSP Section 2 item # 18A (REC_OASDI_INSURANCE) schema def updated as not a required field
**Description:**

Per the coding instructions, SSP Section 2 item # 18A (REC_OASDI_INSURANCE) is not a required field. 

2887p1

However, we missed setting this to not required, as shown below: 
https://github.com/raft-tech/TANF-app/blob/abd50999a5d9c793dc30b7daec096059fc1a3d56/tdrs-backend/tdpservice/parsers/schema_defs/ssp/m5.py#L251

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] SSP Section 2 item # 18A (REC_OASDI_INSURANCE) is not a required field. 
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] Update codebase to account for this item number not being a required field. 
- [ ] _Task 2_
- [ ] _Task 3_
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

- _Note 1_
- _Note 2_
- _Note 3_

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

- [workbook](https://gorafttech.sharepoint.com/:x:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/Parsing/data/ssp/TDP_Validation_SSP_Section2.xlsx?d=we5e1fbb5fde343438ae28cb6a3b6363c&csf=1&web=1&e=QgfQbb) updated

- _Doc 2_

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- _Open Question 1_
- _Open Question 2_

### PR DESCRIPTION
**Pull request by atrimpe**
## Summary of Changes
Altered required attribute to False in m5.py. 
Pull request closes #2887 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] SSP Section 2 item # 18A (REC_OASDI_INSURANCE) is not a required field.
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/parsers/schema_defs/ssp/m5.py [hunk 1]
```diff
@@ -88,12 +88,6 @@
                     result_field="CITIZENSHIP_STATUS",
                     result_function=validators.isInLimits(1, 3),
                 ),
-                validators.if_then_validator(
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/parsers/schema_defs/ssp/m5.py]: was removing the postparsing validator part of the ticket requirements?

**Review by jtimpe — APPROVED** [review]
one question, but otherwise approved

**Review by raftmsohani — APPROVED** [review]
LGTM!

File: tdrs-backend/tdpservice/parsers/schema_defs/ssp/m5.py (hunk 1 above)
Comment (atrimpe) [reply tdrs-backend/tdpservice/parsers/schema_defs/ssp/m5.py]: My bad, missed the "of a child" part of the reqs. Putting back and fixing test case.

File: tdrs-backend/tdpservice/parsers/test/test_parse.py [hunk 2]
```diff
@@ -1146,6 +1146,26 @@ def test_parse_ssp_section4_file(ssp_section4_file, dfs):
...
+
+@pytest.mark.django_db()
+def test_parse_ssp_section2_rec_oadsi_file(ssp_section2_rec_oadsi_file, dfs):
+    """Test parsing SSP Section 2 REC_OADSI."""
+    ssp_section2_rec_oadsi_file.year = 2019
+    ssp_section2_rec_oadsi_file.quarter = 'Q1'
+
+    dfs.datafile = ssp_section2_rec_oadsi_file
+
+    parse.parse_datafile(ssp_section2_rec_oadsi_file)
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/parsers/test/test_parse.py]: you didn't do anything wrong, i'd just like to make one point about these tests being more integration-tests than unit tests.

here, we create a whole new data file and send it through the entire parsing routine just to check that a single field validator works. it'd be nice to have a test that only checks that `required=False` does what it's supposed to, then we can be confident in uses of `required=False` going forward without these large/complex integration tests (and not have to write a new test every time a arbitrary field's `required` value changes). smaller unit tests are also more resilient against side effects or changes in other parts of the system.

this is the primary intent behind #2641

**Review by jtimpe — APPROVED** [review]

File: tdrs-backend/tdpservice/parsers/test/test_parse.py (hunk 2 above)
Comment (elipe17) [reply tdrs-backend/tdpservice/parsers/test/test_parse.py]: If we leave the test, I think you should also query appropriate record and verify that the field is `None`.

**Review by elipe17 — APPROVED** [review]

File: tdrs-backend/tdpservice/parsers/test/data/ssp_section2_rec_oadsi_file.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+HEADER20184C24   SSP1EU
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/parsers/test/data/ssp_section2_rec_oadsi_file.txt]:
```suggestion
HEADER20184C24   SSP1ED
```

minor suggestion given #2899

**Review by ADPennington — APPROVED** [review]
@atrimpe lgtm 🚀 

testing notes [here](https://hackmd.io/Ban2OmoQT_unHqLLLngUpg#2887---18A-REC_OASDI_INSURANCE-not-mandatory)

**Pull request merged**